server/mailet/integration-testing/src/test/java/org/apache/james/mailets/SubAddressingTest.java

```diff
@@ -97,43 +97,106 @@ void tearDown() throws IOException {
     void subAddressedEmailShouldBeDeliveredInINBOXWhenSpecifiedFolderDoesNotExist(@TempDir File temporaryFolder) throws Exception {
         setup(temporaryFolder);
 
-        //do not create mailbox
+        // do not create mailbox
+        assertThat(testIMAPClient.sendCommand("GETACL " + TARGETED_MAILBOX)).contains("Mailbox not found");
 
         sendSubAddressedMail();
         awaitSubAddressedMail(MailboxConstants.INBOX);
     }
 
     @Test
-    void subAddressedEmailShouldBeDeliveredInINBOXWhenNoRights(@TempDir File temporaryFolder) throws Exception {
+    void subAddressedEmailShouldBeDeliveredInINBOXWhenNobodyHasRight(@TempDir File temporaryFolder) throws Exception {
         setup(temporaryFolder);
 
         // create mailbox
-        testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
-            .login(RECIPIENT, PASSWORD)
-            .create(TARGETED_MAILBOX);
+        testIMAPClient.sendCommand("CREATE " + TARGETED_MAILBOX);
+        assertThat(testIMAPClient.sendCommand("GETACL " + TARGETED_MAILBOX)).contains("OK GETACL");
```

Comment:
Tests are supposed to get only a single assertion. THis allow making clear what is tested.

If intermediate states are unsure, this likely means a distinct test shall be added, covering explicitly the effects of SETACL. Also those very valuable tests have not much to do with subadressing and shall be done in dedicated location (RIghts.script in mpt-core ?)

Reply:
yes you're right about the extra assertions, they are just for me "to be sure" but they are technically useless.

about the dedicated location, I can't find Rights.script or mpt-core ; would mpt/impl/imap-mailbox/inmemory/src/test/java/org/apache/james/mpt/imapmailbox/inmemory/InMemoryPositionRightTest.java be ok ?

Reply:
For the third time I share the link of the existing test file for Right management on Apache James 

https://github.com/apache/james-project/blob/master/mpt/impl/imap-mailbox/core/src/main/resources/org/apache/james/imap/scripts/Rights.test